wurst/systems/spawns/ResourceSpawnsUtils.wurst

```diff
@@ -0,0 +1,197 @@
+package ResourceSpawnsUtils
+
+import Composer
+
+// Standard library imports:
+import HashMap
+import Interpolation
+import TerrainUtils
+
+// Local imports:
+import Game
+import GameTimer
+import GameConfig
+import GeometryUtils
+import Pool
+
+function getRandomAllowedPoint(Pool<rect> allowedRegions, region restrainedSpawnRegion) returns vec2
+    let spawnRect = allowedRegions.random()

```

Comment:
This should have a global that controls the maximum number of points picked and an error message if it fails that many times. This should be a large number (e.g. 1000) that safeguards against an infinite loop when the region contains no valid points, which could happen during terrain changes and be very annoying to debug without this message.